### PR TITLE
fix broken doc strings

### PR DESCRIPTION
**Pull request by rkimballn1**
Doc strings must be `/// \blah` or they don't count as doc strings from the compilers point of view.
If something has `/// \deprecated` then it must also have the deprecated attribute or `NGRAPH_DEPRECATED`.
Once the doc strings were the correct format a couple of errors were reported and fixed.

### CODE REVIEW

File: src/ngraph/op/batch_norm.hpp [hunk 1]
```diff
@@ -26,39 +27,40 @@ namespace ngraph
...
+            ///
+            /// SHAPE DETAILS:
+            ///   gamma:     must have rank 1, with the same span as input's channel axis.
+            ///   beta:      must have rank 1, with the same span as input's channel axis.
+            ///   input:     must have rank >= 2.  The second dimension represents the channel axis
+            ///              and must have a span of at least 1.
+            ///   output[0]: shall have the same shape as 'input'.
+            ///   output[1]: shall have rank 1, with the same span as input's channel axis.
+            ///   output[2]: shall have rank 1, with the same span as input's channel axis.
+            NGRAPH_DEPRECATED("Use another constructor")
```
Comment (aprocter) [inline src/ngraph/op/batch_norm.hpp]: This is a bit vague... maybe `Use BatchNormTraining::BatchNormTraining(input, gamma, beta, epsilon)

File: src/ngraph/op/batch_norm.hpp [hunk 2]
```diff
@@ -85,37 +87,38 @@ namespace ngraph
...
+            ///
+            /// SHAPE DETAILS:
+            ///   gamma:    must have rank 1, with the same span as input's channel axis.
+            ///   beta:     must have rank 1, with the same span as input's channel axis.
+            ///   input:    must have rank >= 2. The second dimension represents the channel axis and
+            ///             must have a span of at least 1.
+            ///   mean:     must have rank 1, with the same span as input's channel axis.
+            ///   variance: must have rank 1, with the same span as input's channel axis.
+            ///   output:   shall have the same shape as 'input'.
+            NGRAPH_DEPRECATED("Use another constructor")
```
Comment (aprocter) [inline src/ngraph/op/batch_norm.hpp]: Again, specify the alternative

File: src/ngraph/op/batch_norm.hpp [hunk 3]
```diff
@@ -157,7 +160,8 @@ namespace ngraph
                                       std::shared_ptr<Node> delta,
                                       double epsilon);
 
-            // \deprecated
+            NGRAPH_DEPRECATED_DOC
+            NGRAPH_DEPRECATED("Use another constructor")
```
Comment (aprocter) [inline src/ngraph/op/batch_norm.hpp]: Again, specify the alternative
